lightstep/sdk/metric/internal/viewstate/base_instrument.go

```diff
@@ -26,79 +26,88 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 )
 
+// storageHolder is a generic struct for holding one storage and one
+// auxiliary field.  Storage will be one of the aggregators.  The
```

Comment:
Should we enforce the type constraint that storage has to be an aggregator and not an "any"

Reply:
I added a package-level comment, since this is practically going to be a FAQ. The Storage does satisfy an interface, but the receiver methods we want would be pointer methods. The use of a paired Methods struct works around this so that the SDK can control memory layout.

Comment will appear in #174 with the Aggregators, where the Methods interface is defined.
```
// package aggregator defines the interface used between the SDK and
// various kinds of aggregation.
//
// Note, about the use of Go-1.18 generics!  This code makes use of a
// pattern that gives the SDK fine control over memory layout via the
// use of a `Storage any` type constraint.  The SDK is expected to
// pair the allocated `Storage any` value with corresponding `Methods`
// type, which provides pointer-receiver methods that operate on one
// or two `Storage` objects.
//
// This means there are places where a Storage object may appear in a
// generic type.  Where that happens without the corresponding
// Methods, it means the Storage is opaque at that level in the code.
```